publisher/moloco.js

```diff
@@ -27,24 +27,25 @@ function constructNativeAdDiv(nativeAd, nativeAdDiv) {
   return nativeAdDiv;
 }
 
-var MolocoSDK = function(data) {
-    this.endpoint = "//bidfnt-asia.adsmoloco.com/adserver?mobile_web=1";
-    this.adUnit = data["ad_unit"];
-    this.adType = data["ad_type"];
-    this.containerId = data["container_id"];
-    this.fallbackContainerId = data["fallback_container_id"];
-    this.idfa = data["idfa"];
-    this.bundle = data["bundle"];
-    this.country = data["country_iso"];
-    this.width = data["width"];
-    this.height = data["height"];
-    this.extra = encodeURIComponent(data["extra"] || "");
+var MolocoSDK = function (data) {
+  this.endpoint = "//adservfnt-asia.adsmoloco.com/adserver?mobile_web=1";
+  this.adUnit = data["ad_unit"];
+  this.adType = data["ad_type"];
+  this.containerId = data["container_id"];
+  this.fallbackContainerId = data["fallback_container_id"];
+  this.idfa = data["idfa"];
+  this.bundle = data["bundle"];
+  this.country = data["country_iso"];
+  this.width = data["width"];
+  this.height = data["height"];
+  this.extra = encodeURIComponent(data["extra"] || "");
+  this.os = data["os"];
 }
 
 // Send ad request to Moloco.
-MolocoSDK.prototype.requestAd = function(adDiv) {
+MolocoSDK.prototype.requestAd = function (adDiv) {
   const orientation = window.innerWidth > window.innerHeight ? "l" : "p";
-  let url = this.endpoint + "&id=" + this.adUnit + "&udid=ifa:" + this.idfa + "&bundle=" + this.bundle + "&iso=" + this.country + "&w=" + this.width + "&h=" + this.height + "&o=" + orientation + "&ufid=" + uuidv4() + "&x=" + this.extra;
+  let url = this.endpoint + "&id=" + this.adUnit + "&udid=ifa:" + this.idfa + "&bundle=" + this.bundle + "&iso=" + this.country + "&w=" + this.width + "&h=" + this.height + "&o=" + orientation + "&ufid=" + uuidv4() + "&x=" + this.extra + "os=" + this.os;
```

Comment:
[MUST]
I think you need to add `&` in front of "os="

Reply:
Yup, I updated the code with checking the os parameter is undefined.